File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        unordered_map<int, int> count;
        long long goodPairs = 0;
        long long totalPairs = 0;
        for (int i = 0; i < nums.size(); i++){
            totalPairs += i;
            int index = nums[i] - i;
            goodPairs += count.contains(index) ? count[index] : 0;
            if (count.contains(index))
                count[index] += 1;
            else
                count[index] = 1;
        }

        return totalPairs - goodPairs;
    }
};

File: 2448-count-number-of-bad-pairs/count-number-of-bad-pairs.java
class Solution {
    public long countBadPairs(int[] nums) {
        HashMap<Integer, Integer> count = new HashMap<Integer, Integer>();
        long goodPairs = 0;
        long totalPairs = 0;
        for (int i = 0; i < nums.length; i++){
            totalPairs += i;
            int index = nums[i] - i;
            goodPairs += count.containsKey(index) ? count.get(index) : 0;
            if (count.containsKey(index))
                count.put(index, count.get(index) + 1);
            else
                count.put(index, 1);
        }

        return totalPairs - goodPairs;  
    }
}